### ISSUE TITLE
Cache mutation bug

### ISSUE THREAD

**Issue by bparees** [openshift/origin#15324]
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/3256/

```
I0719 02:12:14.503031   10463 stateful_set.go:452] Syncing StatefulSet default/mysql with 1 pods
CACHE *v1beta1.StatefulSet[4] ALTERED!
```

`Cache mutations are not flakes. They are P0 bugs and you found one. Open it up!`

was run against post-rebase code.

**Comment by bparees** [comment]
Note there are two other failures in that run also, that make it appear as if the server went away:

```
=== BEGIN TEST CASE ===
hack/test-cmd.sh:126: executing 'oc delete project 'cmd-admin'' expecting success
FAILURE after 0.684s: hack/test-cmd.sh:126: executing 'oc delete project 'cmd-admin'' expecting success: the command returned the wrong error code
There was no output from the command.
Standard error from the command:
The connection to the server 172.17.0.2:28443 was refused - did you specify the right host or port?
=== END TEST CASE ===
```

probably unrelated, but also probably rebase related?

**Comment by mfojtik** [comment]
seems like upstream bug?

**Comment by deads2k** [comment]
> seems like upstream bug?

Yes, but it will block our CI until it's fixed.

**Comment by deads2k** [comment]
> probably unrelated, but also probably rebase related?

Same run?  The cache mutation kills the server.

**Comment by mfojtik** [comment]
attempt to fix this: https://github.com/openshift/origin/pull/15328

**Comment by bparees** [comment]
Yeah same run.

Ben Parees | OpenShift

**Comment by liggitt** [comment]
weird, the diff showed kind and apiVersion fields populated, and nothing else changed...

I bet passing an object to an encoder sets those fields.

**Comment by deads2k** [comment]
> attempt to fix this: #15328

Assigned you since you have a pull.

**Comment by kargakis** [comment]
This is fixed and @mfojtik said he will open a follow-up issue upstream to track investigation on this.
